apps/bfd-pipeline/bfd-pipeline-rif-extract/src/main/java/gov/cms/bfd/pipeline/rif/extract/s3/DataSetTestUtilities.java

```diff
@@ -30,6 +31,29 @@
  * without having to delve into classpath dark arts.
  */
 public class DataSetTestUtilities {
+  /**
+   * @param s3Client the {@link AmazonS3} client to use
+   * @return a new, random {@link Bucket} for use in an integration test
+   */
+  public static Bucket createTestBucket(AmazonS3 s3Client) {
+    String username = System.getProperty("user.name");
+    if (username == null || username.isEmpty()) username = "anonymous";
+    username.replaceAll("@", "-");
+    username.replaceAll("\\\\", "-");
+    int randomId = new Random().nextInt(100000);
+    String bucketName = String.format("bb-test-%s-%d", username, randomId);
+
+    Bucket bucket = s3Client.createBucket(bucketName);
+    /*
+     * Note: S3's API is eventually consistent, so any calls to use this new bucket will
+     * intermittently fail for a brief period of time. The only solution to this is to retry all of
```

Comment:
You could also use a [waiter](https://aws.amazon.com/blogs/developer/waiters-in-the-aws-sdk-for-java/) here to pause until the bucket is available. I think it would be:

```
s3Client.waiters()
        .bucketExists()
        .run(new HeadBucketRequest().withBucketName(bucketName));
```

Reply:
Interesting!

Does that cover the eventual consistency issue, or will it just verify that whichever endpoint it happens to connect to has the bucket?

Reply:
It will cover DNS and S3 metadata service eventual consistency.